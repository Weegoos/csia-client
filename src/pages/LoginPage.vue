<template>
  <div class="q-pa-md bg-black">
    <section class="text-white q-mb-md">
      <p class="text-h5 text-bold">Log in to the system</p>
      <p class="text-body2">
        Please enter your credentials to access your account
      </p>
    </section>
    <section>
      <q-input
        v-model="email"
        placeholder="Enter your email"
        stack-label
        class="text-white input q-my-md"
        color="white"
        input-class="text-white"
        label-class="text-white"
        input-style="color: white;"
        rounded
        outlined
      />
      <q-input
        v-model="password"
        :type="isPwd ? 'password' : 'text'"
        placeholder="Enter your password"
        stack-label
        class="text-white input q-my-md"
        color="white"
        input-class="text-white"
        label-class="text-white"
        input-style="color: white;"
        rounded
        outlined
      >
        <template v-slot:append>
          <q-icon
            :name="isPwd ? 'visibility_off' : 'visibility'"
            class="cursor-pointer"
            color="grey"
            @click="isPwd = !isPwd"
          />
        </template>
      </q-input>
      <q-btn
        color="green-4"
        style="width: 100%"
        rounded
        no-caps
        label="Sign in with email"
        @click="login"
      />
      <q-btn
        color="black"
        style="width: 100%"
        rounded
        no-caps
        class="q-my-md"
        label="Sign up with email"
        @click="pushToRegistration"
      />
    </section>
  </div>
</template>

<script setup>
import { ref } from "vue";
import { useRouter } from "vue-router";

// global variables
const router = useRouter();

const isPwd = ref(true);
const email = ref("");
const password = ref("");
const slide = ref("fullname");
const lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";

const login = () => {
  console.log("Registering with", {
    email: email.value,
    password: password.value,
  });
};

const pushToRegistration = () => {
  router.push("/registration");
};
</script>

<style scoped>
.input {
  background-color: #30322e;
  border-radius: 54px;
}
</style>
