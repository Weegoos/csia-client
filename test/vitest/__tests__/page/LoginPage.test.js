import { installQuasarPlugin } from "@quasar/quasar-app-extension-testing-unit-vitest";
import { mount } from "@vue/test-utils";
import { createPinia } from "pinia";
import { Quasar } from "quasar";
import LoginPage from "src/pages/LoginPage.vue";
import { describe, expect, it } from "vitest";
import messages from "../../../../src/i18n/en-US/index";
import { createI18n } from "vue-i18n";
installQuasarPlugin();

const i18n = createI18n({
  legacy: false,
  locale: "ru-RU",
  messages,
});

const pinia = createPinia();

describe("tests for LoginPage", () => {
  const wrapper = mount(LoginPage, {
    global: {
      plugins: [pinia, Quasar, i18n],
    },
  });
  it("should find loginMainText data-testid", () => {
    const loginMainText = wrapper.find('[data-testid="loginMainText"]');
    expect(loginMainText.exists()).toBe(true);
  });

  it("should find loginSubText data-testid", () => {
    const loginSubText = wrapper.find('[data-testid="loginSubText"]');
    expect(loginSubText.exists()).toBe(true);
  });

  it("should find emailInput data-testid", () => {
    const emailInput = wrapper.find('[data-testid="emailInput"]');
    expect(emailInput.exists()).toBe(true);

    emailInput.setValue("example@mail.ru");
    expect(emailInput.element.value).toBe("example@mail.ru");
  });

  it("should find passwordInput data-testid", () => {
    const passwordInput = wrapper.find('[data-testid="passwordInput"]');
    expect(passwordInput.exists()).toBe(true);

    passwordInput.setValue("examplePassword");
    expect(passwordInput.element.value).toBe("examplePassword");
  });
});
