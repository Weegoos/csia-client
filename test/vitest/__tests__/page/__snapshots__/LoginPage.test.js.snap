// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for LoginPage > should find loginMainText data-testid 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <p
    class="text-h5 text-bold"
    data-testid="loginMainText"
    data-v-16d8eba4=""
  >
     Log in to the system 
  </p>,
}
`;

exports[`tests for LoginPage > should find loginSubText data-testid 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <p
    class="text-body2"
    data-testid="loginSubText"
    data-v-16d8eba4=""
  >
     Please enter your credentials to access your account 
  </p>,
}
`;
